sql/src/test/java/io/crate/integrationtests/disruption/routing/PrimaryAllocationIT.java

```diff
@@ -0,0 +1,310 @@
+
+        logger.info("--> starting node that reuses data folder with the up-to-date shard");
+        internalCluster().startDataOnlyNode(Settings.EMPTY);
+        assertBusy(() -> assertTrue(client().admin().cluster().prepareState().get().getState()
+            .getRoutingTable().index(indexName).allPrimaryShardsUnassigned()));
+    }
+
+    public void testNotWaitForQuorumCopies() throws Exception {
+        logger.info("--> starting 3 nodes");
+        List<String> nodes = internalCluster().startNodes(3);
+        logger.info("--> creating index with 1 primary and 2 replicas");
+        execute("create table t (x string) clustered into " + randomIntBetween(1, 3) +
+                " shards with (number_of_replicas = 2, \"write.wait_for_active_shards\" = 1)");
+        ensureGreen();
+        execute("insert into t values ('value1')");
+        logger.info("--> removing 2 nodes from cluster");
+        internalCluster().stopRandomNode(InternalTestCluster.nameFilter(nodes.get(1), nodes.get(2)));
+        internalCluster().stopRandomNode(InternalTestCluster.nameFilter(nodes.get(1), nodes.get(2)));
+        internalCluster().restartRandomDataNode();
+        logger.info("--> checking that index still gets allocated with only 1 shard copy being available");
+        ensureYellow();
+        execute("select * from t");
+        assertThat(response.rowCount(), is(1L));
+    }
+
+    /**
+     * This test ensures that for an unassigned primary shard that has a valid shard copy on at least one node,
+     * we will force allocate the primary shard to one of those nodes, even if the allocation deciders all return
+     * a NO decision to allocate.
+     */
+    public void testForceAllocatePrimaryOnNoDecision() throws Exception {
+        logger.info("--> starting 1 node");
+        final String node = internalCluster().startNode();
+        logger.info("--> creating index with 1 primary and 0 replicas");
+        execute("create table t (x string) clustered into 1 shards with (number_of_replicas = 0)");
+        logger.info("--> update the settings to prevent allocation to the data node");
+        execute("set global cluster.routing.allocation.exclude._name = '" + node + "'");
+        logger.info("--> full cluster restart");
+        internalCluster().fullRestart();
+        logger.info("--> checking that the primary shard is force allocated to the data node despite being blocked by the exclude filter");
+        ensureGreen();
+        assertEquals(1, client().admin().cluster().prepareState().get().getState()
+                            .routingTable().index(indexName).shardsWithState(ShardRoutingState.STARTED).size());
+    }
+
+    /**
+     * This test asserts that replicas failed to execute resync operations will be failed but not marked as stale.
+     */
+    @TestLogging("_root:DEBUG, org.elasticsearch.cluster.routing.allocation:TRACE, org.elasticsearch.cluster.action.shard:TRACE," +
```

Comment:
debug traces?

Reply:
generally (root) debug, but some particulars on traces. this is what ES uses so I used the same as in case it fails I assume it's useful to have some (presumably relevant) output.